docs/api_v2.md

```diff
@@ -0,0 +1,105 @@
+- Memory capacity (in bytes)
+- Maximum supported CPU frequency (in kHz)
+- Available filesystems: major, minor numbers and capacity (in bytes)
+- Network devices: mac addresses, MTU, and speed (if available)
+- Machine topology: Nodes, cores, threads, per-node memory, and caches
+
+The actual object is the marshalled JSON of the `MachineInfo` struct found in [info/v1/machine.go](../info/v1/machine.go)
+
+## Attributes
+
+Attributes endpoint provides hardware and software attributes of the running machine.
+The resource name for attributes is:
+`/api/v2.0/attributes`
+
+Hardware information includes all information covered by machine endpoint. Software information include version of cAdvisor, kernel, docker, and underlying OS.
+
+The actual object is the marshalled JSON of the `Attributes` struct found in [info/v2/machine.go](../info/v2/machine.go)
+
+## Container Stats
+The resource name for container stats information is:
+`/api/v2.0/stats/<container identifier>`
+
+### Stats request options
+
+Stats support following options in the request:
+- `type`: describes the type of identifier. Supported values are `name`(default) and `docker`. `name` implies that the identifier is an absolute container name. `docker` implies that the identifier is a docker id.
+- `recursive`: Option to specify if stats for subcontainers of the requested containers should also be reported. Default is false.
+- `count`: Number of stats samples to be reported. Default is 64.
+
+### Container name
+
+When container identifier is of type `name`, the identifier is interpreted as the absolute container name. Naming follows the lmctfy convention. For example:
+
+| Container Name       | Resource Name                             |
+|----------------------|-------------------------------------------|
+| /                    | /api/v2.0/containers/                     |
+| /foo                 | /api/v2.0/containers/foo                  |
+| /docker/2c4dee605d22 | /api/v2.0/containers/docker/2c4dee605d22  |
+
+Note that the root container (`/`) contains usage for the entire machine. All Docker containers are listed under `/docker`. Also, `type=name` is not required in the examples above as `name` is the default type.
+
+### Docker Containers
+
+When container identifier is of type `docker`, the identifier is interpreted as docker id. For example:
+
+
+| Docker container     | Resource Name                             |
+|----------------------|-------------------------------------------|
+| All docker containers| /api/v2.0/stats?type=docker&recursive=true|
+| clever_colden        | /api/v2.0/stats/clever_colden?type=docker |
+| 2c4dee605d22         | /api/v2.0/stats/2c4dee605d22?type=docker  |
+
+The Docker name can be either the UUID or the short name of the container. It returns the information of the specified container(s).
+
+Note that `recursive` is only valid when docker root is specified. It is used to get stats for all docker containers.
```

Comment:
Ahh got it. May be add a sentence about 'recursive' option for regular
cgroup stats as well?

On Mon, Mar 23, 2015 at 12:38 PM, Rohit Jnagal <notifications@github.com>
wrote:

> In docs/api_v2.md
> <https://github.com/google/cadvisor/pull/611#discussion_r26971391>:
>
> > +Note that the root container (`/`) contains usage for the entire machine. All Docker containers are listed under `/docker`. Also, `type=name` is not required in the examples above as `name` is the default type.
> > +
> > +### Docker Containers
> > +
> > +When container identifier is of type `docker`, the identifier is interpreted as docker id. For example:
> > +
> > +
> > +| Docker container     | Resource Name                             |
> > +|----------------------|-------------------------------------------|
> > +| All docker containers| /api/v2.0/stats?type=docker&recursive=true|
> > +| clever_colden        | /api/v2.0/stats/clever_colden?type=docker |
> > +| 2c4dee605d22         | /api/v2.0/stats/2c4dee605d22?type=docker  |
> > +
> > +The Docker name can be either the UUID or the short name of the container. It returns the information of the specified container(s).
> > +
> > +Note that `recursive` is only valid when docker root is specified. It is used to get stats for all docker containers.
>
> It still works with regular combination when container name is used. Only
> for docker containers, we don't accept docker container id and
> recursive=true.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/google/cadvisor/pull/611/files#r26971391>.
>